core-plugins/src/main/java/co/cask/hydrator/plugin/common/StructuredToTextTransformer.java

```diff
@@ -18,31 +18,42 @@
 package co.cask.hydrator.plugin.common;
 
 import co.cask.cdap.api.data.format.StructuredRecord;
-import co.cask.hydrator.common.RecordConverter;
+import co.cask.cdap.api.data.schema.Schema;
 
 import java.io.IOException;
+import java.util.List;
 
 /**
  * Creates Single String from StructuredRecord
  */
-public class StructuredToTextTransformer extends RecordConverter<StructuredRecord, String> {
-
+public class StructuredToTextTransformer {
   private final String delimiter;
-  private static final String FIELD_NAME = "record";
 
   public StructuredToTextTransformer(String delimiter) {
     this.delimiter = delimiter;
   }
 
-  @Override
-  public String transform(StructuredRecord structuredRecord,
-                          co.cask.cdap.api.data.schema.Schema schema) throws IOException {
-    String recordString = "";
-    for (co.cask.cdap.api.data.schema.Schema.Field field : schema.getFields()) {
-      Object recordField = structuredRecord.get(field.getName());
-      String fieldValue = recordField == null ? "" : recordField.toString();
-      recordString += fieldValue + delimiter;
+  /**
+   * appends the structured record fields as strings
+   * separated by a delimiter. if the fields value is null they are empty strings.
+   * returns concatenated string representing the record.
+   * @param structuredRecord
+   * @return String
+   * @throws IOException
+   */
+  public String transform(StructuredRecord structuredRecord) throws IOException {
+    StringBuilder builder = new StringBuilder();
+    List<Schema.Field> fields = structuredRecord.getSchema().getFields();
+    builder.append(getFieldAsString(structuredRecord.get(fields.get(0).getName())));
+
+    for (int index = 1; index < fields.size(); index++) {
```

Comment:
I think its always an ArrayList, but it's still better to use an iterator rather than get(index) since there is no guarantee that it's O(1).